### ISSUE TITLE
React to window events

### ISSUE THREAD

**Issue by theRealProHacker** [theRealProHacker/Positron#26]
Window events are events like 
- window resize
- window move
- mouse move
- mouse down
- mouse up
- key down and up
- etc.

Ideally, these events should be handled by a custom event manager who also detects events like 
- mouse
  - click,
  - double click, 
  - drag and swipe
- key
  - press, 
  - double-press, 
  - hold, 
  - with key modifiers.

The user should then be able to create callbacks for events, that are automatically called.
```python
# this is just an example and I can say it will definitely not be the actual API
from Positron.Events import EventManager
EventManager.on("click", do_something) # do_something can be sync or async and it gets the event passed.
EventManager.on("key-press", lambda event: sys.exit() if event.key.is("esc") else True)
```

**Comment by theRealProHacker** [comment]
Implemented some events in #37

**Comment by theRealProHacker** [comment]
This is mainly implemented with the continuously growing `EventManager`
